src/help.jl

```diff
@@ -3,8 +3,17 @@
 import Markdown
 import REPL
 
+if isdefined( REPL.TerminalMenus, :default_terminal)
+  default_terminal = REPL.TerminalMenus.default_terminal
+else
+  # before Julia 1.12
+  function default_terminal()
+    return REPL.TerminalMenus.terminal
+  end
+end
+
 function gap_help_string(topic::String, onlyexact::Bool = false,
-    term::REPL.Terminals.TTYTerminal = REPL.TerminalMenus.terminal;
+    term::REPL.Terminals.TTYTerminal = default_terminal();
```

Comment:
```suggestion
    term::REPL.Terminals.TTYTerminal = default_terminal()
```
As far as I can tell, `term` is only used in one spot, in the call to `REPL.TerminalMenus.request` below. So perhaps we should just replace `term` there by `default_terminal();`

Reply:
Here `term` is an argument for the function `gap_help_string`.
The idea behind the complicated syntax is to make this function testable, by providing input from a terminal, see `test/help.jl`.
(I think the fact that we found the problem with Julia nightly shows that this test is useful.)

Reply:
Ahhhhh of course you are 100% right; I completely misread the code and missed that term is an argument